lib/http/request_util.js

```diff
@@ -1,37 +1,213 @@
+const LoggingUtil = require('../logger/logging_util');
 const sfParams = require('../constants/sf_params');
 
+// Initial whitelist for attributes - they will be described with values
+exports.DEFAULT_ATTRIBUTES_DESCRIBING_REQUEST_WITH_VALUES = [
+  'baseUrl',
+  'path',
+  sfParams.paramsNames.SF_REQUEST_ID,
+  sfParams.paramsNames.SF_REQUEST_GUID,
+  sfParams.paramsNames.SF_WAREHOUSE_NAME,
+  sfParams.paramsNames.SF_DB_NAME,
+  sfParams.paramsNames.SF_SCHEMA_NAME,
+];
+
+// Initial blacklist for attributes - described as present/not present only
+exports.DEFAULT_ATTRIBUTES_DESCRIBING_REQUEST_WITHOUT_VALUES = [
+  sfParams.paramsNames.SF_TOKEN
+];
+
 /**
- * Should work with not yet parsed options as well (before calling prepareRequestOptions method).
+ * Describes a request based on its options.
+ * Should work with not-yet-parsed options as well (before calling prepareRequestOptions method).
  *
- * @param requestOptions - object representing the request data with top-level keys
- * @returns {string}
+ * @param {Object} requestOptions - Object representing the request data with top-level keys.
+ * @param {Object} [options] - Options for describing attributes.
+ * @param {Array<string>} [options.additionalAttributesDescribedWithValues]
+ * @param {Array<string>} [options.overrideAttributesDescribedWithValues]
+ * @param {Array<string>} [options.additionalAttributesDescribedWithoutValues]
+ * @param {Array<string>} [options.overrideAttributesDescribedWithoutValues]
+ * @returns {string} A string representation of the request data.
  */
-exports.describeRequestFromOptions = function (requestOptions) {
-  return exports.describeRequestData({ method: requestOptions.method, url: requestOptions.url, guid: requestOptions.params?.[sfParams.paramsNames.SF_REQUEST_GUID] });
+exports.describeRequestFromOptions = function (
+  requestOptions,
+  {
+    additionalAttributesDescribedWithValues = [],
+    overrideAttributesDescribedWithValues,
+    additionalAttributesDescribedWithoutValues = [],
+    overrideAttributesDescribedWithoutValues
+  } = {}
+) {
+  const describingAttributesWithValues = mergeAttributeLists(
+    exports.DEFAULT_ATTRIBUTES_DESCRIBING_REQUEST_WITH_VALUES,
```

Comment:
When you need to use these values, it is better to exclude them to consts not directly add to exports.